spec/lib/pedicel/validator_spec.rb

```diff
@@ -0,0 +1,513 @@
+
+      expect(subject).to receive(:hex?).and_return(false)
+      expect(hex_sha256?('d1bc8d3ba4afc7e109612cb73acbdddac052c93025aa1f82942edabb7deb82a1')).to be false
+    end
+
+    it 'false for non-sha256 that are hex encoded' do
+      expect(hex_sha256?('aedf')).to be false
+      expect(hex_sha256?('1bc8d3ba4afc7e109612cb73acbdddac052c93025aa1f82942edabb7deb82a1')).to be false
+      expect(hex_sha256?('Fd1bc8d3ba4afc7e109612cb73acbdddac052c93025aa1f82942edabb7deb82a1')).to be false
+    end
+  end
+
+  describe '.pan?' do
+    def pan?(x); subject.pan?(x); end
+
+    it 'true for valid PANs' do
+      expect(pan?('123456789012')).to be true
+      expect(pan?('1234567890123456789')).to be true
+      expect(pan?('1234567890123456')).to be true
+      expect(pan?('1000000000000000')).to be true
+    end
+
+    it 'false if PAN contains anything but digits' do
+      expect(pan?('1a34567890123456')).to be false
+      expect(pan?('1 34567890123456')).to be false
+      expect(pan?('1_34567890123456')).to be false
+      expect(pan?('134567890123456 ')).to be false
+      expect(pan?(' 134567890123456')).to be false
+    end
+
+    it 'false if PAN starts with a zero' do
+      expect(pan?('0234567890123456')).to be false
+    end
+
+    it 'false if PAN is shorter than 12 digits' do
+      expect(pan?('12345678901')).to be false
+    end
+
+    it 'false if PAN is longer than 19 digits' do
+      expect(pan?('12345678901234567890')).to be false
+    end
+  end
+
+  describe '.yymmdd?' do
+    def yymmdd?(x); subject.yymmdd?(x); end
+
+    it 'is true for valid dates' do
+      expect(yymmdd?('180228')).to be true
+      expect(yymmdd?('200229')).to be true
+    end
+
+    it 'is true for invalid dates that are still date-like' do
+      expect(yymmdd?('170229')).to be true
+      expect(yymmdd?('179901')).to be true
+    end
+
+    it 'is false for non-numeric "dates"' do
+      expect(yymmdd?('17ja19')).to be false
+    end
+  end
+
+  describe '.eci?' do
+    def eci?(x); subject.eci?(x); end
+
+    it 'is true for valid ECIs' do
+      expect(eci?('05')).to be true
+      expect(eci?('06')).to be true
+      expect(eci?('07')).to be true
+    end
+
+    it 'is false for ECIs of wrong length' do
+      expect(eci?('005')).to be false
+      expect(eci?('5')).to   be false
+      expect(eci?(' 05')).to be false
+    end
+
+    it 'is false for non-numeric ECIs' do
+      expect(eci?('  ')).to be false
+      expect(eci?('ab')).to be false
+      expect(eci?('__')).to be false
+    end
+  end
+
+  describe '.ec_public_key?' do
+    def ec_public_key?(x); subject.ec_public_key?(x); end
+
+    it 'is true for valid EC public keys' do
+      expect(ec_public_key?(valid_ec_public_key)).to be true
+    end
+
+    it 'is false for invalid EC public keys' do
+      expect(ec_public_key?('validBase64ButInvalidEcPublicKey')).to be false
+    end
+
+    it 'uses base64? predicate to ensure that it is base 64' do
```

Comment:
`s/base 64/base64/`

Reply:
Hmm. I'll write it "Base64" (https://en.wikipedia.org/wiki/Base64) when writing text.

Likewise, "SHA-256" (https://en.wikipedia.org/wiki/SHA-2) rather than "sha 256"/"sha256".